app/controllers/uploads_controller.rb

```diff
@@ -5,7 +5,7 @@ class UploadsController < ApplicationController
 
   def index
     @uploads = Upload.where(data_source_id: @data_source.id)
-      .page(params[:page].to_i).per(20)
+      .page(params[:page].to_i).per(20).order(created_at: :desc)
```

Comment:
I believe this should be order then page since page and per will only return the 20 you are looking at.